apps/deploy-web/src/components/new-deployment/NewDeploymentContainer.tsx

```diff
@@ -115,9 +115,9 @@ export function NewDeploymentContainer() {
     <Layout isLoading={isLoadingTemplates} isUsingSettings isUsingWallet containerClassName="pb-0">
       <div className="flex w-full items-center">{activeStep !== null && <CustomizedSteppers activeStep={activeStep} />}</div>
 
-      {activeStep === 0 && <TemplateList setSelectedTemplate={setSelectedTemplate} setEditedManifest={setEditedManifest} />}
+      {activeStep === 0 && <TemplateList />}
       {activeStep === 1 && (
-        <ManifestEdit selectedTemplate={selectedTemplate as TemplateCreation} editedManifest={editedManifest as string} setEditedManifest={setEditedManifest} />
+        <ManifestEdit selectedTemplate={selectedTemplate as TemplateCreation} setSelectedTemplate={setSelectedTemplate} editedManifest={editedManifest as string} setEditedManifest={setEditedManifest} />
```

Comment:
I think this type cast is deceiving and may cause issues. `ManifestEdit` should accept it as optional and handle the case when it's not set.

```suggestion
        <ManifestEdit selectedTemplate={selectedTemplate} setSelectedTemplate={setSelectedTemplate} editedManifest={editedManifest as string} setEditedManifest={setEditedManifest} />
```

Reply:
This was not my commits.
I have added only setSelectedTemplate={setSelectedTemplate}.
Anyway, how about below:

      {activeStep === 1 && selectedTemplate && (
        <ManifestEdit
          selectedTemplate={selectedTemplate}
          setSelectedTemplate={setSelectedTemplate}
          editedManifest={editedManifest as string}
          setEditedManifest={setEditedManifest}
        />
      )}

Reply:
Yea, I understand, as I said some of my comments may be treated as refactoring. It's fine not to fulfil them.